### DISCUSSION TITLE
Created a websocket with Flask-io and gevent whenever sent a request to server it open and closes the connection immediately

### DISCUSSION THREAD

**Discussion by Mr-Dj-Python** [miguelgrinberg/Flask-SocketIO#1821] [Q&A] [answered]
### extension.py file
socketio = SocketIO(async_mode="gevent")

## flask_app. py file
   socketio.init_app(app, cors_allowed_origins="*")

## run server via 
 gunicorn -k geventwebsocket.gunicorn.workers.GeventWebSocketWorker --bind localhost:5005 flask_app:app --workers=1 --threads=300

whenever I send a request to the server it opens a connection and closes immediately.

@miguelgrinberg please suggest something

![Screenshot from 2022-04-27 18-34-10](https://user-images.githubusercontent.com/72373042/165525703-0d9d8eb2-9466-4d6f-a5f6-3dd316ad8d1d.png)
![Screenshot from 2022-04-27 18-33-55](https://user-images.githubusercontent.com/72373042/165525712-9f00cdcf-b7cf-4099-806b-981209e173b4.png)

**Comment by miguelgrinberg** [comment, marked as answer]
Start by initializing the extension in a single place. If you want to use a direct initialization, then pass all your arguments to the `SocketIO()` constructor. If you prefer the two-step initialization, then initialize `SocketIO()` with no arguments, and pass all your arguments to `init_app()`.

**Reply by Mr-Dj-Python** [reply]
@miguelgrinberg thanks for quick response 
yes I tried that too still have the same issue

**Reply by miguelgrinberg** [reply]
Also use a Socket.IO client to make a connection.

**Reply by Mr-Dj-Python** [reply]
so if I use the normal WebSocket npm library it won't work with it?

**Reply by miguelgrinberg** [reply]
Well, no. This is a Socket.IO server, not a WebSocket server.

**Reply by Mr-Dj-Python** [reply]
Noted let me try with that thanks

**Reply by Mr-Dj-Python** [reply]
@miguelgrinberg thanks you it is working now
